File: devel/benchmark.py
'''*

    Fornux C++ Superset -- Benchmark.
    
 '''


import time
import os
import resource


class list_node:
    def __init__(self):
        p = 0



def test():
    # Proper cyclic memory leaks management:
    before = int(round(time.time() * 1000000000))
    
    i = 0
    while True:
        i += 1
        
        # cycle
        p = list_node()
        p.p = list_node()
        p.p.p = list_node()
        p.p.p.p = p
            
        if (i) > 500000:
            break

    # stats
    after = int(round(time.time() * 1000000000))
    
    ru_maxrss = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
    print("Python, %li, %f, %li" % ((after - before), i * 1000000000.0 / (after - before), ru_maxrss))

    return 0


test()

File: devel/benchmark.java
/**

    Fornux C++ Superset -- Benchmark.
    
 */


import java.util.*;


class list_node
{
    public list_node p;
}


class Test 
{
    public static void main(String[] args) 
    {
        // Proper cyclic memory leaks management:
        long before, after;
        before = System.nanoTime();

        int i = 0;
        for (; ; ++ i)
        {
            // cycle
            list_node p = new list_node();
            p.p = new list_node();
            p.p.p = new list_node();
            p.p.p.p = p;
            
            if (i > 500000)
                break;
        }
        
        // stats
        after = System.nanoTime();

        long ru_maxrss = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory();

        System.out.printf("Java, %d, %f, %d\n", (after - before), i * 1000000000.0 / (after - before), ru_maxrss);
    }
}
